handling returned slice of interfaces

### The following program `sample.go` triggers an unexpected result

```go
package main

type Option interface {
    apply(*T)
}

type T struct {
    s string
}

type opt struct {                                                               
    name string                                                                 
}                                                                               
                                                                                
func (o *opt) apply(t *T) {                                                     
    println(o.name)                                                             
}                                                                               
                                                                                
func BuidOptions() []Option {                                                   
    return []Option{                                                            
        &opt{"opt1"},                                                           
        &opt{"opt2"},                                                           
    }                                                                           
}                                                                               
                                                                                
func NewT(name string, options ...Option) *T {
    t := &T{name}
    for _, opt := range options {
        opt.apply(t)
    }
    return t
}

func main() {                                                                   
    t := NewT("hello", BuidOptions()...)                                        
    println(t.s)                                                                
}
```


### Expected result

```console
$ go run ./sample.go
opt1
opt2
hello
```


### Got

```console
$ yaegi ./sample.go
./sample.go:27:8: panic
./sample.go:35:21: panic
run: runtime error: invalid memory address or nil pointer dereference
goroutine 1 [running]:
runtime/debug.Stack()
	runtime/debug/stack.go:24 +0x88
github.com/traefik/yaegi/interp.(*Interpreter).Execute.func1(0x140000a9ae0)
	github.com/traefik/yaegi/interp/program.go:131 +0x8c
panic({0x10326c5e0, 0x10383be20})
	runtime/panic.go:1038 +0x21c
github.com/traefik/yaegi/interp.runCfg.func1(0x14000368160, 0x140000a1680, 0x0)
	github.com/traefik/yaegi/interp/run.go:193 +0x164
panic({0x10326c5e0, 0x10383be20})
	runtime/panic.go:1038 +0x21c
github.com/traefik/yaegi/interp.runCfg.func1(0x140003682c0, 0x1400009f680, 0x0)
	github.com/traefik/yaegi/interp/run.go:193 +0x164
panic({0x10326c5e0, 0x10383be20})
	runtime/panic.go:1038 +0x21c
github.com/traefik/yaegi/interp.getMethodByName.func1(0x140003682c0)
	github.com/traefik/yaegi/interp/run.go:1921 +0x1c0
github.com/traefik/yaegi/interp.runCfg(0x1400009f680, 0x140003682c0, 0x1400009e000, 0x140000a1320)
	github.com/traefik/yaegi/interp/run.go:201 +0x268
github.com/traefik/yaegi/interp.call.func9(0x14000368160)
	github.com/traefik/yaegi/interp/run.go:1387 +0x7c0
github.com/traefik/yaegi/interp.runCfg(0x140000a1680, 0x14000368160, 0x140000a0a20, 0x0)
	github.com/traefik/yaegi/interp/run.go:201 +0x268
github.com/traefik/yaegi/interp.(*Interpreter).run(0x14000366000, 0x140000a0a20, 0x14000368000)
	github.com/traefik/yaegi/interp/run.go:120 +0x3d4
github.com/traefik/yaegi/interp.(*Interpreter).Execute(0x14000366000, 0x1400008daa0)
	github.com/traefik/yaegi/interp/program.go:157 +0x1e0
github.com/traefik/yaegi/interp.(*Interpreter).eval(0x14000366000, {0x14000202540, 0x1b9}, {0x16d6536d6, 0x18}, 0x0)
	github.com/traefik/yaegi/interp/interp.go:615 +0x98
github.com/traefik/yaegi/interp.(*Interpreter).EvalPath(0x14000366000, {0x16d6536d6, 0x18})
	github.com/traefik/yaegi/interp/interp.go:508 +0xe8
main.runFile(0x14000366000, {0x16d6536d6, 0x18}, 0x0)
	github.com/traefik/yaegi/cmd/yaegi/run.go:153 +0xb0
main.run({0x140001c0010, 0x1, 0x1})
	github.com/traefik/yaegi/cmd/yaegi/run.go:116 +0xe14
main.main()
	github.com/traefik/yaegi/cmd/yaegi/yaegi.go:144 +0x384
```


### Yaegi Version

v0.11.1

### Additional Notes

This issue is also related to #1172

--- a/_test/issue-1326.go
+++ b/_test/issue-1326.go
@@ -1,0 +1,42 @@
+package main
+
+type Option interface {
+	apply(*T)
+}
+
+type T struct {
+	s string
+}
+
+type opt struct {
+	name string
+}
+
+func (o *opt) apply(t *T) {
+	println(o.name)
+}
+
+func BuildOptions() []Option {
+	return []Option{
+		&opt{"opt1"},
+		&opt{"opt2"},
+	}
+}
+
+func NewT(name string, options ...Option) *T {
+	t := &T{name}
+	for _, opt := range options {
+		opt.apply(t)
+	}
+	return t
+}
+
+func main() {
+	t := NewT("hello", BuildOptions()...)
+	println(t.s)
+}
+
+// Output:
+// opt1
+// opt2
+// hello

--- a/interp/run.go
+++ b/interp/run.go
@@ -1151,6 +1151,7 @@ func call(n *node) {
 	child := n.child[1:]
 	tnext := getExec(n.tnext)
 	fnext := getExec(n.fnext)
+	hasVariadicArgs := n.action == aCallSlice // callSlice implies variadic call with ellipsis.
 
 	// Compute input argument value functions.
 	for i, c := range child {
@@ -1166,7 +1167,7 @@ func call(n *node) {
 			numOut := c.child[0].typ.rtype.NumOut()
 			for j := 0; j < numOut; j++ {
 				ind := c.findex + j
-				if !isInterfaceSrc(arg) || isEmptyInterface(arg) {
+				if hasVariadicArgs || !isInterfaceSrc(arg) || isEmptyInterface(arg) {
 					values = append(values, func(f *frame) reflect.Value { return f.data[ind] })
 					continue
 				}
@@ -1179,7 +1180,7 @@ func call(n *node) {
 			cc0 := c.child[0]
 			for j := range cc0.typ.ret {
 				ind := c.findex + j
-				if !isInterfaceSrc(arg) || isEmptyInterface(arg) {
+				if hasVariadicArgs || !isInterfaceSrc(arg) || isEmptyInterface(arg) {
 					values = append(values, func(f *frame) reflect.Value { return f.data[ind] })
 					continue
 				}
@@ -1195,8 +1196,7 @@ func call(n *node) {
 			switch {
 			case isEmptyInterface(arg):
 				values = append(values, genValue(c))
-			case isInterfaceSrc(arg) && n.action != aCallSlice:
-				// callSlice implies variadic call with ellipsis, do not wrap in valueInterface.
+			case isInterfaceSrc(arg) && !hasVariadicArgs:
 				values = append(values, genValueInterface(c))
 			case isInterfaceBin(arg):
 				values = append(values, genInterfaceWrapper(c, arg.rtype))
@@ -1926,7 +1926,7 @@ func getMethodByName(n *node) {
 			}
 			return next
 		}
-		m, li := val.node.typ.lookupMethod(name)
+		m, li := typ.lookupMethod(name)
 		if m == nil {
 			panic(n.cfgErrorf("method not found: %s", name))
 		}

